plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -658,5 +681,18 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
     private const val POINT_COUNT = "point_count"
     private const val CLUSTER_TEXT_LAYER_ID = "mapbox-android-cluster-text-layer"
     private val DEFAULT_TEXT_FIELD = get("point_count")
+
+    private var workerThread: HandlerThread? = null
+    private var workerHandler: Handler? = null
+
+    /**
+     * Clean and stop the workerThread for updating source
+     */
+    fun cleanupWorkerThread() {
+      workerHandler?.removeCallbacksAndMessages(null)
+      workerThread?.quit()
```

Comment:
Should be followed up with `workerThread.join` I believe like it's done here
https://github.com/mapbox/mapbox-maps-android/blob/1e3215006be349a19d1ea5c07fd728e55f71a103/sdk/src/main/java/com/mapbox/maps/renderer/WorkerHandlerThread.kt#L29-L37